### PR TITLE
Remove proxy from evm.

### PR DESCRIPTION
**Pull request by Cloud10240**

### CODE REVIEW

**Comment by rappie** [comment]
Is this related to the proxy error?

https://github.com/fuzzland/ityfuzz/issues/64

**Comment by shouc** [comment]
yes

File: Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ name = "ityfuzz"
 harness = false
 
 [features]
-default = ["cmp", "dataflow", "evm", "print_txn_corpus", "full_trace", ]
+default = ["cmp", "dataflow", "evm", "print_txn_corpus", "full_trace"]
```
Comment (shouc) [inline Cargo.toml:15]: do not remove the ", " here. It will break CI logic

File: src/evm/onchain/endpoints.rs [hunk 2]
```diff
@@ -66,7 +75,7 @@ impl Chain {
     pub fn get_chain_rpc(&self) -> String {
         match self {
             Chain::ETH => "https://eth.llamarpc.com",
-            Chain::BSC => "https://bsc.llamarpc.com",
+            Chain::BSC => "https://quiet-omniscient-daylight.bsc.quiknode.pro/88e8c7767457e3c8a2dc8ba889e18f63bd2a5b5d/",
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:78]: dont push our private endpoint, please undo last commit and use force push to overwrite it

File: src/evm/onchain/endpoints.rs [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
 use std::cell::RefCell;
-use std::collections::HashMap;
 use std::fmt::Debug;
 use std::panic;
 use std::rc::Rc;
 use std::str::FromStr;
 use std::sync::Arc;
 use std::time::Duration;
 
+const MAX_HOPS: u32 = 5; // Assuming the value of MAX_HOPS
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:25]: MAX_HOPS should be 2

File: src/evm/onchain/endpoints.rs [hunk 4]
```diff
@@ -389,37 +445,28 @@ impl OnChainConfig {
             }
         );
         println!("fetching abi from {}", endpoint);
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:447]: Can you check for a DELEGATECALL contract to see whether the tool can also propagate the ABI of proxy target to proxy during fuzzing on the fly? 

(e.g., we should have PayLuckWinner (0x88b6be2d) in corpus with target as  0x0955A73D014F0693aC7B53CFe77706dAb02b3ef9 in the corpus)

File: src/evm/onchain/endpoints.rs [hunk 5]
```diff
@@ -693,30 +678,351 @@ impl OnChainConfig {
...
+                continue;
+            }
+            let next_hops = self.get_all_hops(&i.next, network, block, hop + 1, known);
+            hops.extend(next_hops);
+        }
+
+        hops
+    }
+
+    fn get_pegged_next_hop(&self, token: &str, network: &str, block: &str) -> PairData {
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:890]: add unit test for this

File: src/evm/onchain/endpoints.rs [hunk 6]
```diff
@@ -693,30 +678,351 @@ impl OnChainConfig {
...
+            }
+        };
+
+        let reserve1 = &result[3..67];
+        let reserve2 = &result[67..131];
+
+        (reserve1.into(), reserve2.into())
+    }
+
+    fn get_all_hops(
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:860]: add unit test for this

File: src/evm/onchain/endpoints.rs [hunk 7]
```diff
@@ -693,30 +678,351 @@ impl OnChainConfig {
...
+                        initial_reserves_0: "".to_string(),
+                        initial_reserves_1: "".to_string(),
+                    });
+                }
+            }
+        }
+        next_tokens
+    }
+
+    fn get_pair_pegged(&self, token: &str, network: &str, block: &str) -> Vec<PairData> {
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:725]: add unit test for this

File: src/evm/onchain/endpoints.rs [hunk 8]
```diff
@@ -693,30 +678,351 @@ impl OnChainConfig {
...
-                    let data = res.text().unwrap().trim().to_string();
-                    if data == "0,0" {
-                        None
-                    } else {
-                        let parts: Vec<u32> =
-                            data.split(",").map(|x| x.parse::<u32>().unwrap()).collect();
-                        assert_eq!(parts.len(), 2);
-                        Some((parts[0], parts[1]))
-                    }
+    fn get_pair(&self, token: &str, network: &str, block: &str) -> Vec<PairData> {
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:681]: add unit test for this

File: src/evm/onchain/endpoints.rs [hunk 9]
```diff
@@ -566,118 +580,89 @@ impl OnChainConfig {
     }
 
     pub fn fetch_uniswap_path(&self, token_address: H160) -> TokenContext {
```
Comment (shouc) [inline src/evm/onchain/endpoints.rs:582]: add unit test for this

**Pull request merged**